components/builder-api/src/server/db.rs

```diff
@@ -0,0 +1,20 @@
+use actix_web::actix::{Actor, Addr, SyncArbiter, SyncContext};
+use db::config::DataStoreCfg;
+use diesel::prelude::PgConnection;
+use diesel::r2d2::ConnectionManager;
+use r2d2::Pool;
+
+pub struct DbPool(pub Pool<ConnectionManager<PgConnection>>);
+
+impl Actor for DbPool {
+    type Context = SyncContext<Self>;
+}
+
+pub fn init(config: DataStoreCfg) -> Addr<DbPool> {
+    let manager = ConnectionManager::<PgConnection>::new(config.to_string());
+    let pool = Pool::builder()
+        .max_size(config.pool_size)
+        .build(manager)
+        .expect("Failed to create pool.");
+    SyncArbiter::start(4, move || DbPool(pool.clone()))
```

Comment:
config.db_workers instead of 4?

Reply:
I want to add this to builder-db but can't until we shift things around so we don't have so many reliant services on the current config structure.